tests/unit/s2n_handshake_test.c

```diff
@@ -434,6 +434,42 @@ int main(int argc, char **argv)
         }
     }
 
+    /* Ensure that a handshake can be performed after all file descriptors are closed */
+    {
+        /* A fork is created to ensure that closing file descriptors (like stdout) won't impact
+         * other tests.
+         */
+        pid_t pid = fork();
+        if (pid == 0) {
+            long max_file_descriptors = sysconf(_SC_OPEN_MAX);
+            EXPECT_SUCCESS(max_file_descriptors);
+            EXPECT_TRUE(max_file_descriptors <= INT_MAX);
+            for (int fd = 0; fd <= max_file_descriptors; fd++) {
+                close(fd);
+            }
```

Comment:
I know this will probably never matter, but I still don't think this is right. Let's pretend we're dealing with uint8_ts here instead of int.
UINT8_MAX is 255. But uint8_t can also be 0 (and 0 is a valid fd-- that's stdin). So there are actually 256 different possibles values for a uint8_t.
So you'd want to check <= UINT8_MAX + 1, not <= UINT8_MAX. For a valid value you could check <= UINT8_MAX tho.

So I think this should be:
```suggestion
            EXPECT_SUCCESS(max_file_descriptors);
            EXPECT_TRUE(max_file_descriptors <= INT_MAX + 1);
            for (int fd = 0; fd < max_file_descriptors; fd++) {
                close(fd);
            }
```
Or maybe this would be clearer:
```suggestion
            EXPECT_SUCCESS(max_file_descriptors);
            long max_fd = MIN(INT_MAX, max_file_descriptors - 1);
            for (int fd = 0; fd < max_fd; fd++) {
                close(fd);
            }
```